subsys/net/ip/net_pkt.c

```diff
@@ -1767,7 +1767,9 @@ static void clone_pkt_attributes(struct net_pkt *pkt, struct net_pkt *clone_pkt)
 {
 	net_pkt_set_family(clone_pkt, net_pkt_family(pkt));
 	net_pkt_set_context(clone_pkt, net_pkt_context(pkt));
+#if defined(CONFIG_NET_IP)
 	net_pkt_set_ip_hdr_len(clone_pkt, net_pkt_ip_hdr_len(pkt));
+#endif
```

Comment:
Instead of adding these `#ifdef` here and there, we could add them to `net_pkt_set_ip_hdr_len()` 
so something like
```
void net_pkt_set_ip_hdr_len(pkt, len)
{
#if defined(CONFIG_NET_IP)
set len here....
#endif
}
```

Reply:
Fixed here as well as for the corresponding getter.